superset-frontend/src/pages/SavedQueryList/index.tsx

```diff
@@ -233,16 +232,31 @@ function SavedQueryList({
   };
 
   const copyQueryLink = useCallback(
-    (id: number) => {
-      copyTextToClipboard(() =>
-        Promise.resolve(`${window.location.origin}/sqllab?savedQueryId=${id}`),
-      )
-        .then(() => {
-          addSuccessToast(t('Link Copied!'));
-        })
-        .catch(() => {
-          addDangerToast(t('Sorry, your browser does not support copying.'));
+    async (savedQuery: SavedQueryObject) => {
+      try {
+        const payload = {
+          dbId: savedQuery.db_id,
+          name: savedQuery.label,
+          schema: savedQuery.schema,
```

Comment:
We need to store the catalog as well, for databases that support it.
```suggestion
          catalog: savedQuery.catalog,
          schema: savedQuery.schema,
```

Reply:
Hey, so currently the catalog property does not exist on the SavedQueryObject type [https://github.com/apache/superset/blob/master/superset-frontend/src/views/CRUD/types.ts](Here).

Reply:
Ok, let me add it before you merge this, we need it.

Reply:
okay sounds good.

Reply:
OK, I tested and it you add `catalog` to `superset-frontend/src/views/CRUD/types.ts` it will be populated automatically by the hook, so you just need to add:

```js
catalog: string | null;
```

to `SavedQueryObject`.

Also, it needs to be added to the Schemas and Types as well, because if a database supports multiple catalogs (like BigQuery or Trino) the query with just the schema is not enough information.